### PR TITLE
refactor(toml): Expose surce/spans for VirtualManifests

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

This is a follow up to #13593, expanding support from `Manifest` to `VirtualManifest` as well.

This also does other clean up along the way in preparation for making a more explicit `resolve_toml` phase.

### CODE REVIEW

**Comment by epage** [comment]
CI is blocked until `master` gets a version update post beta-branch.

**Comment by weihanglo** [comment]
> CI is blocked until `master` gets a version update post beta-branch.

Will do that today

File: src/cargo/util/toml/mod.rs [hunk 1]
```diff
@@ -1362,7 +1365,12 @@ fn to_virtual_manifest(
             bail!("virtual manifests must be configured with [workspace]");
         }
     };
+    let resolved_toml = original_toml.clone();
```
[the remark this answers is not in the document]
Comment (epage) [reply src/cargo/util/toml/mod.rs]: We do a small normalization later on.  Over time, I will be doing more.

**Review by weihanglo — APPROVED** [review]

**Pull request merged**